docs/rfc/rfc-2.adoc

```diff
@@ -689,24 +625,96 @@ func BuildTransactionHexString(
-
-	return txBuilder.String(), nil
+	return tx, nil
 }
 ```
 <1> These two come from `fundingInfo().utxoOutpoint`, which has the transaction
     id as the first 32 little-endian bytes and the previous output index as
     its last 4 little-endian bytes.
 <2> Comes from `fundingInfo().utxoValue`, a little-endian unsigned int.
 <3> Derived as part of the <<signing-protocol>>.
-<4> Computed via the <<signing-protocol>>.
-<5> Available on the signer directly (see
+<4> e.g. &chaincfg.MainNetParams or TestNet3Params
+
+
+[[sighash-construction]]
+==== Construction of Sighash for Signing
+
+The sighash is constructed as the double-SHA256 of the preimage. The preimage
+is defined in BIP143. Fortunately, once the dummy unsigned transaction is
+available, btcutil provides the tooling to build the sighash. The sample
+function call below shows how to construct the sighash once an unsigned
+transaction is available from above the function in
+<<transaction-construction>>. This sighash is the data then signed by the
+signer group to produce a valid transaction (see the section on
+<<final-transaction-construction>> for the rest of the process once the
+signature has been computed):
+
+```go
+// import "github.com/btcsuite/btcd/txscript"
+
+// unsignedTransaction := ConstructUnsignedTransaction(...)
+sighashBytes, err := txscript.CalcWitnessSigHash(
+  previousOutputScriptCode, // from PublicKeyToP2WPKHScriptCode
+  txscript.NewTxSigHashes(unsignedTransaction),
+  txscript.SigHashAll,
+  unsignedTransaction,
+  0, // sighash for the first input; there is only one input anyway
+  previousOutputValue,
+)
+```
+
+[[final-transaction-construction]]
+==== Construction of Final Transaction
+
+The final transaction can be constructed from the unsigned transaction
+produced in <<transaction-construction>>, the signature of that transaction's
+sighash from <<sighash-construction>>, and the public key of the signing group.
+The sample function below shows the various inputs as well as how they can
+be combined to produce the final transaction hex string that can then be
+submitted to the chain:
+
+```go
+func BuildSignedTransactionHexString(
+  unsignedTransaction *wire.MsgTx, // <1>
+	signature *ecdsa.Signature, // <2>
+	publicKey *ecdsa.PublicKey, // <3>
+) (string, error) {
+  // For safety's sake, work on a deep copy, as mutations follow.
+  signedTransaction := unsignedTransaction.Copy()
+
+	// Construct witness script.
+	signatureDERBytes := (&btcec.Signature{R: signature.R, S: signature.S}).Serialize()
+	hashType := uint8(txscript.SigHashAll)
+
+	// The witness signature field is the DER signature followed by the hash type.
+	signatureForWitnessBuffer := &bytes.Buffer{}
+	signatureForWitnessBuffer.Write(signatureDERBytes)
+	binary.Write(signatureForWitnessBuffer, binary.LittleEndian, hashType)
+
+  // The witness is for the first input, since this is known to be a
+  // single-input transaction.
+  signedTransaction.TxIn[0].Witness = [][]byte{
```

Comment:
Could construct a `wire.TxWitness` object instead of `[][]byte`? For reference please see our old code: 
https://github.com/keep-network/keep-ecdsa/blob/bef6601428605a61d26d98c1e66cda751a049f6c/pkg/btc/witness.go#L15-L32

Reply:
Going to need to dig into the type juggling going on in there. I guess you can assign structs to slices somehow?

Anyway, yes, this does seem Clearer™.

Reply:
Oh. This is actually the exact same thing. `wire.TxWitness` is [a type alias for `[][]byte`](https://github.com/btcsuite/btcd/blob/v0.21.0-beta/wire/msgtx.go#L241). The rest does effectively the same thing as I was doing already, just with a slightly modified set of function calls. I'll still switch over to it as I think it's more concise, and `wire.TxWitness` makes the intent clearer than `[][]byte`.

Reply:
Done.